lib/SDIR/Translate/translateToSDFG.cpp

```diff
@@ -0,0 +1,1597 @@
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// LoadOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateLoadToSDFG(LoadOp &op, JsonEmitter &jemit) {
+  return success();
+}
+
+//===----------------------------------------------------------------------===//
+// StoreOp
+//===----------------------------------------------------------------------===//
+
+LogicalResult translateStoreToSDFG(StoreOp &op, JsonEmitter &jemit) {
+  jemit.startObject();
+  jemit.printKVPair("type", "MultiConnectorEdge");
+
+  jemit.startNamedObject("attributes");
+  jemit.startNamedObject("data");
+  jemit.printKVPair("type", "Memlet");
+  jemit.startNamedObject("attributes");
+
+  jemit.printKVPair("volume", 1);
+  jemit.printKVPair("num_accesses", 1);
+
+  if (GetAccessOp aNode = dyn_cast<GetAccessOp>(op.arr().getDefiningOp())) {
+    jemit.printKVPair("data", aNode.getName());
+  } else {
+    mlir::emitError(op.getLoc(), "Array must be defined by a GetAccessOp");
+    return failure();
+  }
+
+  jemit.startNamedObject("subset");
+  jemit.printKVPair("type", "Range");
+  jemit.startNamedList("ranges");
+
+  if (ArrayAttr syms = op->getAttr("indices").cast<ArrayAttr>()) {
+    for (Attribute sym : syms.getValue()) {
+      if (StringAttr sym_str = sym.cast<StringAttr>()) {
+        jemit.startObject();
+        jemit.printKVPair("start", sym_str.getValue());
+        jemit.printKVPair("end", sym_str.getValue());
+        jemit.printKVPair("step", 1);
+        jemit.printKVPair("tile", 1);
+        jemit.endObject();
+      } else {
+        mlir::emitError(op.getLoc(), "'indices' must consist of StringAttr");
+        return failure();
+      }
+    }
+  } else {
+    mlir::emitError(op.getLoc(), "'indices' must be an ArrayAttr");
+    return failure();
+  }
+
+  jemit.endList();   // ranges
+  jemit.endObject(); // subset
+
+  jemit.startNamedObject("dst_subset");
+  jemit.printKVPair("type", "Range");
+  jemit.startNamedList("ranges");
+
+  if (ArrayAttr syms = op->getAttr("indices").cast<ArrayAttr>()) {
+    for (Attribute sym : syms.getValue()) {
+      if (StringAttr sym_str = sym.cast<StringAttr>()) {
+        jemit.startObject();
+        jemit.printKVPair("start", sym_str.getValue());
+        jemit.printKVPair("end", sym_str.getValue());
+        jemit.printKVPair("step", 1);
+        jemit.printKVPair("tile", 1);
+        jemit.endObject();
+      } else {
+        mlir::emitError(op.getLoc(), "'indices' must consist of StringAttr");
+        return failure();
+      }
+    }
+  } else {
+    mlir::emitError(op.getLoc(), "'indices' must be an ArrayAttr");
+    return failure();
+  }
+
+  jemit.endList();   // ranges
+  jemit.endObject(); // dst_subset
+
+  jemit.endObject(); // attributes
+  jemit.endObject(); // data
+  jemit.endObject(); // attributes
+
+  if (sdir::CallOp call = dyn_cast<sdir::CallOp>(op.val().getDefiningOp())) {
```

Comment:
seems to be irrelevant to the store itself

Reply:
How so?
If the store operation represents a `tasklet -> access node` edge, then we need the ID of the tasklet to refer to the specific tasklet

Reply:
Right, that makes sense! Please add a comment.